### ISSUE TITLE
Latest version (0.5.3) gives Error Starting Kernel - Internal Server Error

### ISSUE THREAD

**Issue by seembha** [renan-r-santos/pixi-kernel#39]
I have JupyterLab installed in a global environment with the latest (version 0.5.3) pixi-kernel installed
```
PS C:\Users\seembha\Projects\Python\test\src\test> pixi global list
Global environments as specified in 'C:\Users\seembha\.pixi\manifests\pixi-global.toml'
└── python
    ├─ dependencies: python 3.13.1, jupyterlab 4.3.4, pixi-kernel 0.5.3
    └─ exposes: jlpm, python, pydoc, jupyter-labhub, jupyter-labextension, jupyter-lab, idle, pythonw
```
And, a separate `test` Pixi python project environment with ipykernel installed.
```
PS C:\Users\seembha\Projects\Python\test\src\test> pixi info
System
------------
      Pixi version: 0.39.0
          Platform: win-64
  Virtual packages: __win=0=0
                  : __archspec=1=x86_64_v4
         Cache dir: C:\Users\seembha\AppData\Local\rattler\cache
      Auth storage: C:\Users\seembha\.rattler\credentials.json
  Config locations: No config files found

Global
------------
           Bin dir: C:\Users\seembha\.pixi\bin
   Environment dir: C:\Users\seembha\.pixi\envs
      Manifest dir: C:\Users\seembha\.pixi\manifests\pixi-global.toml

Project
------------
              Name: test
           Version: 0.1.0
     Manifest file: C:\Users\seembha\Projects\Python\test\pyproject.toml
      Last updated: 23-12-2024 00:09:30

Environments
------------
       Environment: default
          Features: default
          Channels: conda-forge
  Dependency count: 2
      Dependencies: python, ipykernel
 PyPI Dependencies: test
  Target platforms: win-64


PS C:\Users\seembha\Projects\Python\test\src\test> pixi list
Package             Version       Build               Size       Kind   Source
asttokens           3.0.0         pyhd8ed1ab_1        27.5 KiB   conda  asttokens
bzip2               1.0.8         h2466b09_7          53.6 KiB   conda  bzip2
ca-certificates     2024.12.14    h56e8100_0          153.7 KiB  conda  ca-certificates
colorama            0.4.6         pyhd8ed1ab_1        26.4 KiB   conda  colorama
comm                0.2.2         pyhd8ed1ab_1        11.8 KiB   conda  comm
cpython             3.13.1        py313hd8ed1ab_102   45.9 KiB   conda  cpython
debugpy             1.8.11        py313h5813708_0     3.4 MiB    conda  debugpy
decorator           5.1.1         pyhd8ed1ab_1        13.7 KiB   conda  decorator
exceptiongroup      1.2.2         pyhd8ed1ab_1        20 KiB     conda  exceptiongroup
executing           2.1.0         pyhd8ed1ab_1        27.7 KiB   conda  executing
importlib-metadata  8.5.0         pyha770c72_1        28 KiB     conda  importlib-metadata
ipykernel           6.29.5        pyh4bbf305_0        117 KiB    conda  ipykernel
ipython             8.31.0        pyh7428d3b_0        587.3 KiB  conda  ipython
jedi                0.19.2        pyhd8ed1ab_1        823.9 KiB  conda  jedi
jupyter_client      8.6.3         pyhd8ed1ab_1        103.8 KiB  conda  jupyter_client
jupyter_core        5.7.2         pyh5737063_1        56.9 KiB   conda  jupyter_core
krb5                1.21.3        hdf4eb48_0          695.3 KiB  conda  krb5
libexpat            2.6.4         he0c23c2_0          135.8 KiB  conda  libexpat
libffi              3.4.2         h8ffe710_5          41.1 KiB   conda  libffi
liblzma             5.6.3         h2466b09_1          101.9 KiB  conda  liblzma
libmpdec            4.0.0         h2466b09_0          86.6 KiB   conda  libmpdec
libsodium           1.0.20        hc70643c_0          197.6 KiB  conda  libsodium
libsqlite           3.47.2        h67fdade_0          870.4 KiB  conda  libsqlite
libzlib             1.3.1         h2466b09_2          54.2 KiB   conda  libzlib
matplotlib-inline   0.1.7         pyhd8ed1ab_1        14.1 KiB   conda  matplotlib-inline
nest-asyncio        1.6.0         pyhd8ed1ab_1        11.3 KiB   conda  nest-asyncio
openssl             3.4.0         h2466b09_0          8.1 MiB    conda  openssl
packaging           24.2          pyhd8ed1ab_2        58.8 KiB   conda  packaging
parso               0.8.4         pyhd8ed1ab_1        73.5 KiB   conda  parso
pickleshare         0.7.5         pyhd8ed1ab_1004     11.5 KiB   conda  pickleshare
platformdirs        4.3.6         pyhd8ed1ab_1        20 KiB     conda  platformdirs
prompt-toolkit      3.0.48        pyha770c72_1        263.5 KiB  conda  prompt-toolkit
psutil              6.1.0         py313ha7868ed_0     496.6 KiB  conda  psutil
pure_eval           0.2.3         pyhd8ed1ab_1        16.3 KiB   conda  pure_eval
pygments            2.18.0        pyhd8ed1ab_1        856.2 KiB  conda  pygments
python              3.13.1        h071d269_102_cp313  16 MiB     conda  python
python-dateutil     2.9.0.post0   pyhff2d567_1        217.3 KiB  conda  python-dateutil
python_abi          3.13          5_cp313             6.6 KiB    conda  python_abi
pywin32             307           py313h5813708_3     5.8 MiB    conda  pywin32
pyzmq               26.2.0        py313h2100fd5_3     358.9 KiB  conda  pyzmq
six                 1.17.0        pyhd8ed1ab_0        16 KiB     conda  six
stack_data          0.6.3         pyhd8ed1ab_1        26.4 KiB   conda  stack_data
test                0.1.0                                        pypi    (editable)
tk                  8.6.13        h5226925_1          3.3 MiB    conda  tk
tornado             6.4.2         py313ha7868ed_0     845.6 KiB  conda  tornado
traitlets           5.14.3        pyhd8ed1ab_1        107.5 KiB  conda  traitlets
typing_extensions   4.12.2        pyha770c72_1        38.7 KiB   conda  typing_extensions
tzdata              2024b         hc8b5060_0          119.5 KiB  conda  tzdata
ucrt                10.0.22621.0  h57928b3_1          546.6 KiB  conda  ucrt
vc                  14.3          ha32ba9b_23         17.1 KiB   conda  vc
vc14_runtime        14.42.34433   he29a5d6_23         736.6 KiB  conda  vc14_runtime
vs2015_runtime      14.42.34433   hdffcdeb_23         17.2 KiB   conda  vs2015_runtime
wcwidth             0.2.13        pyhd8ed1ab_1        31.8 KiB   conda  wcwidth
zeromq              4.3.5         ha9f60a1_7          2.4 MiB    conda  zeromq
zipp                3.21.0        pyhd8ed1ab_1        21.3 KiB   conda  zipp
```

When I run JupyterLab in the `test` project and select `Python(Pixi)` kernel, I get a window with `Error Starting Kernel - Internal Server Error`. When I revert to pixi-kernel version 0.5.2 in the global environment, everything works as expected. 
The log messages from the PowerShell window are appended below

### pixi-kernel version 0.5.3 log
```
PS C:\Users\seembha\Projects\Python\test\src\test> jupyter-lab.exe
[I 2024-12-23 00:58:21.070 ServerApp] jupyter_lsp | extension was successfully linked.
[I 2024-12-23 00:58:21.074 ServerApp] jupyter_server_terminals | extension was successfully linked.
[I 2024-12-23 00:58:21.080 ServerApp] jupyterlab | extension was successfully linked.
[I 2024-12-23 00:58:21.452 ServerApp] notebook_shim | extension was successfully linked.
[I 2024-12-23 00:58:21.498 ServerApp] notebook_shim | extension was successfully loaded.
[I 2024-12-23 00:58:21.500 ServerApp] jupyter_lsp | extension was successfully loaded.
[I 2024-12-23 00:58:21.501 ServerApp] jupyter_server_terminals | extension was successfully loaded.
[I 2024-12-23 00:58:21.502 LabApp] JupyterLab extension loaded from C:\Users\seembha\.pixi\envs\python\Lib\site-packages\jupyterlab
[I 2024-12-23 00:58:21.502 LabApp] JupyterLab application directory is C:\Users\seembha\.pixi\envs\python\share\jupyter\lab
[I 2024-12-23 00:58:21.503 LabApp] Extension Manager is 'pypi'.
[I 2024-12-23 00:58:21.926 ServerApp] jupyterlab | extension was successfully loaded.
[I 2024-12-23 00:58:21.928 ServerApp] Serving notebooks from local directory: C:\Users\seembha\Projects\Python\test\src\test
[I 2024-12-23 00:58:21.928 ServerApp] Jupyter Server 2.15.0 is running at:
[I 2024-12-23 00:58:21.928 ServerApp] http://localhost:8888/lab?token=d2d98886e62f67dc978f497d71a15d90118bd751b2ac205d
[I 2024-12-23 00:58:21.928 ServerApp]     http://127.0.0.1:8888/lab?token=d2d98886e62f67dc978f497d71a15d90118bd751b2ac205d
[I 2024-12-23 00:58:21.928 ServerApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[C 2024-12-23 00:58:21.985 ServerApp]

    To access the server, open this file in a browser:
        file:///C:/Users/seembha/AppData/Roaming/jupyter/runtime/jpserver-17768-open.html
    Or copy and paste one of these URLs:
        http://localhost:8888/lab?token=d2d98886e62f67dc978f497d71a15d90118bd751b2ac205d
        http://127.0.0.1:8888/lab?token=d2d98886e62f67dc978f497d71a15d90118bd751b2ac205d
[I 2024-12-23 00:58:22.129 ServerApp] Skipped non-installed server(s): bash-language-server, dockerfile-language-server-nodejs, javascript-typescript-langserver, jedi-language-server, julia-language-server, pyright, python-language-server, python-lsp-server, r-languageserver, sql-language-server, texlab, typescript-language-server, unified-language-server, vscode-css-languageserver-bin, vscode-html-languageserver-bin, vscode-json-languageserver-bin, yaml-language-server
[W 2024-12-23 00:58:23.915 LabApp] Could not determine jupyterlab build status without nodejs
[W 2024-12-23 00:58:24.033 ServerApp] 404 GET /api/contents/Untitled.ipynb?content=0&hash=0&1734944304029 (d3cd70ea95e74674a730088af5a0fc90@::1) 1.09ms referer=http://localhost:8888/lab
[W 2024-12-23 00:58:24.034 ServerApp] 404 GET /api/contents/Untitled.ipynb?content=0&hash=0&1734944304029 (::1): No such file or directory: Untitled.ipynb
[W 2024-12-23 00:58:24.035 ServerApp] 404 GET /api/contents/Untitled.ipynb?content=0&hash=0&1734944304029 (d3cd70ea95e74674a730088af5a0fc90@::1) 0.80ms referer=http://localhost:8888/lab
[W 2024-12-23 00:58:24.035 ServerApp] 404 GET /api/contents/Untitled.ipynb?content=0&hash=0&1734944304029 (::1): No such file or directory: Untitled.ipynb
[I 2024-12-23 00:58:27.895 ServerApp] Creating new notebook in
pixi-kernel INFO: JupyterLab provided this value for cwd: C:\Users\seembha\Projects\Python\test\src\test
pixi-kernel INFO: The current working directory is C:\Users\seembha\Projects\Python\test\src\test
[E 2024-12-23 00:58:28.413 ServerApp]
    Traceback (most recent call last):
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\jupyter_client\manager.py", line 87, in wrapper
        out = await method(self, *args, **kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\jupyter_client\manager.py", line 435, in _async_start_kernel
        kernel_cmd, kw = await self._async_pre_start_kernel(**kw)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\jupyter_client\manager.py", line 400, in _async_pre_start_kernel
        kw = await self.provisioner.pre_launch(**kw)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\pixi_kernel\provisioner.py", line 38, in pre_launch
        pixi_environment = await ensure_readiness(
                           ^^^^^^^^^^^^^^^^^^^^^^^
        ...<4 lines>...
        )
        ^
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\pixi_kernel\pixi.py", line 72, in ensure_readiness
        process, stdout, stderr = await subprocess_exec("pixi", "--version", env=env)
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\pixi_kernel\pixi.py", line 36, in subprocess_exec
        process = await create_subprocess_exec(program, *args, stdout=PIPE, stderr=PIPE, **kwargs)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\asyncio\subprocess.py", line 224, in create_subprocess_exec
        transport, protocol = await loop.subprocess_exec(
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
        ...<3 lines>...
            stderr=stderr, **kwds)
            ^^^^^^^^^^^^^^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\asyncio\base_events.py", line 1787, in subprocess_exec
        transport = await self._make_subprocess_transport(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
            protocol, popen_args, False, stdin, stdout, stderr,
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
            bufsize, **kwargs)
            ^^^^^^^^^^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\asyncio\base_events.py", line 534, in _make_subprocess_transport
        raise NotImplementedError
    NotImplementedError
[E 2024-12-23 00:58:28.421 ServerApp]
    Traceback (most recent call last):
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\jupyter_client\multikernelmanager.py", line 232, in _add_kernel_when_ready
        await kernel_awaitable
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\jupyter_core\utils\__init__.py", line 198, in ensure_async
        result = await obj
                 ^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\jupyter_server\services\kernels\kernelmanager.py", line 832, in wrapped_method
        raise err
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\jupyter_server\services\kernels\kernelmanager.py", line 796, in wrapped_method
        out = await method(self, *args, **kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\jupyter_server\services\kernels\kernelmanager.py", line 901, in start_kernel
        return await super().start_kernel(*args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\jupyter_client\manager.py", line 96, in wrapper
        raise e
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\jupyter_client\manager.py", line 87, in wrapper
        out = await method(self, *args, **kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\jupyter_client\manager.py", line 435, in _async_start_kernel
        kernel_cmd, kw = await self._async_pre_start_kernel(**kw)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\jupyter_client\manager.py", line 400, in _async_pre_start_kernel
        kw = await self.provisioner.pre_launch(**kw)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\pixi_kernel\provisioner.py", line 38, in pre_launch
        pixi_environment = await ensure_readiness(
                           ^^^^^^^^^^^^^^^^^^^^^^^
        ...<4 lines>...
        )
        ^
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\pixi_kernel\pixi.py", line 72, in ensure_readiness
        process, stdout, stderr = await subprocess_exec("pixi", "--version", env=env)
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\pixi_kernel\pixi.py", line 36, in subprocess_exec
        process = await create_subprocess_exec(program, *args, stdout=PIPE, stderr=PIPE, **kwargs)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\asyncio\subprocess.py", line 224, in create_subprocess_exec
        transport, protocol = await loop.subprocess_exec(
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
        ...<3 lines>...
            stderr=stderr, **kwds)
            ^^^^^^^^^^^^^^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\asyncio\base_events.py", line 1787, in subprocess_exec
        transport = await self._make_subprocess_transport(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
            protocol, popen_args, False, stdin, stdout, stderr,
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
            bufsize, **kwargs)
            ^^^^^^^^^^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\asyncio\base_events.py", line 534, in _make_subprocess_transport
        raise NotImplementedError
    NotImplementedError
[W 2024-12-23 00:58:28.426 ServerApp] wrote error: 'Internal Server Error'
    Traceback (most recent call last):
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\jupyter_server\services\sessions\handlers.py", line 90, in post
        s_model = await sm.create_session(
                  ^^^^^^^^^^^^^^^^^^^^^^^^
        ...<5 lines>...
        )
        ^
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\jupyter_server\services\sessions\sessionmanager.py", line 285, in create_session
        kernel_id = await self.start_kernel_for_session(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
            session_id, path, name, type, kernel_name
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        )
        ^
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\jupyter_server\services\sessions\sessionmanager.py", line 343, in start_kernel_for_session
        kernel_id = await self.kernel_manager.start_kernel(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        ...<3 lines>...
        )
        ^
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\jupyter_server\services\kernels\kernelmanager.py", line 233, in _async_start_kernel
        kernel_id = await self.pinned_superclass._async_start_kernel(self, **kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\jupyter_client\multikernelmanager.py", line 283, in _async_start_kernel
        raise km.ready.exception()  # type: ignore[misc]
        ^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\jupyter_client\multikernelmanager.py", line 232, in _add_kernel_when_ready
        await kernel_awaitable
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\jupyter_core\utils\__init__.py", line 198, in ensure_async
        result = await obj
                 ^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\jupyter_server\services\kernels\kernelmanager.py", line 832, in wrapped_method
        raise err
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\jupyter_server\services\kernels\kernelmanager.py", line 796, in wrapped_method
        out = await method(self, *args, **kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\jupyter_server\services\kernels\kernelmanager.py", line 901, in start_kernel
        return await super().start_kernel(*args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\jupyter_client\manager.py", line 96, in wrapper
        raise e
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\jupyter_client\manager.py", line 87, in wrapper
        out = await method(self, *args, **kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\jupyter_client\manager.py", line 435, in _async_start_kernel
        kernel_cmd, kw = await self._async_pre_start_kernel(**kw)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\jupyter_client\manager.py", line 400, in _async_pre_start_kernel
        kw = await self.provisioner.pre_launch(**kw)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\pixi_kernel\provisioner.py", line 38, in pre_launch
        pixi_environment = await ensure_readiness(
                           ^^^^^^^^^^^^^^^^^^^^^^^
        ...<4 lines>...
        )
        ^
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\pixi_kernel\pixi.py", line 72, in ensure_readiness
        process, stdout, stderr = await subprocess_exec("pixi", "--version", env=env)
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\pixi_kernel\pixi.py", line 36, in subprocess_exec
        process = await create_subprocess_exec(program, *args, stdout=PIPE, stderr=PIPE, **kwargs)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\asyncio\subprocess.py", line 224, in create_subprocess_exec
        transport, protocol = await loop.subprocess_exec(
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
        ...<3 lines>...
            stderr=stderr, **kwds)
            ^^^^^^^^^^^^^^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\asyncio\base_events.py", line 1787, in subprocess_exec
        transport = await self._make_subprocess_transport(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
            protocol, popen_args, False, stdin, stdout, stderr,
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
            bufsize, **kwargs)
            ^^^^^^^^^^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\asyncio\base_events.py", line 534, in _make_subprocess_transport
        raise NotImplementedError
    NotImplementedError

    The above exception was the direct cause of the following exception:

    Traceback (most recent call last):
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\tornado\web.py", line 1790, in _execute
        result = await result
                 ^^^^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\jupyter_server\auth\decorator.py", line 73, in inner
        return await out
               ^^^^^^^^^
      File "C:\Users\seembha\.pixi\envs\python\Lib\site-packages\jupyter_server\services\sessions\handlers.py", line 108, in post
        raise web.HTTPError(500, str(e)) from e
    tornado.web.HTTPError: HTTP 500: Internal Server Error
[E 2024-12-23 00:58:28.434 ServerApp] {
      "Host": "localhost:8888",
      "Accept": "*/*",
      "Referer": "http://localhost:8888/lab/tree/Untitled.ipynb",
      "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/131.0.0.0 Safari/537.36 Edg/131.0.0.0"
    }
[E 2024-12-23 00:58:28.434 ServerApp] 500 POST /api/sessions?1734944308175 (d3cd70ea95e74674a730088af5a0fc90@::1) 258.66ms referer=http://localhost:8888/lab/tree/Untitled.ipynb
```
### pixi-kernel version 0.5.2 log
```
PS C:\Users\seembha\Projects\Python\test\src\test> jupyter-lab.exe
[I 2024-12-23 00:46:06.837 ServerApp] jupyter_lsp | extension was successfully linked.
[I 2024-12-23 00:46:06.841 ServerApp] jupyter_server_terminals | extension was successfully linked.
[I 2024-12-23 00:46:06.845 ServerApp] jupyterlab | extension was successfully linked.
[I 2024-12-23 00:46:07.212 ServerApp] notebook_shim | extension was successfully linked.
[I 2024-12-23 00:46:07.258 ServerApp] notebook_shim | extension was successfully loaded.
[I 2024-12-23 00:46:07.260 ServerApp] jupyter_lsp | extension was successfully loaded.
[I 2024-12-23 00:46:07.261 ServerApp] jupyter_server_terminals | extension was successfully loaded.
[I 2024-12-23 00:46:07.263 LabApp] JupyterLab extension loaded from C:\Users\seembha\.pixi\envs\python\Lib\site-packages\jupyterlab
[I 2024-12-23 00:46:07.263 LabApp] JupyterLab application directory is C:\Users\seembha\.pixi\envs\python\share\jupyter\lab
[I 2024-12-23 00:46:07.264 LabApp] Extension Manager is 'pypi'.
[I 2024-12-23 00:46:07.684 ServerApp] jupyterlab | extension was successfully loaded.
[I 2024-12-23 00:46:07.686 ServerApp] Serving notebooks from local directory: C:\Users\seembha\Projects\Python\test\src\test
[I 2024-12-23 00:46:07.686 ServerApp] Jupyter Server 2.15.0 is running at:
[I 2024-12-23 00:46:07.686 ServerApp] http://localhost:8888/lab?token=c67283806a5ddce9144a90e0570be7cdc541f716540951c3
[I 2024-12-23 00:46:07.686 ServerApp]     http://127.0.0.1:8888/lab?token=c67283806a5ddce9144a90e0570be7cdc541f716540951c3
[I 2024-12-23 00:46:07.686 ServerApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[C 2024-12-23 00:46:07.735 ServerApp]

    To access the server, open this file in a browser:
        file:///C:/Users/seembha/AppData/Roaming/jupyter/runtime/jpserver-21824-open.html
    Or copy and paste one of these URLs:
        http://localhost:8888/lab?token=c67283806a5ddce9144a90e0570be7cdc541f716540951c3
        http://127.0.0.1:8888/lab?token=c67283806a5ddce9144a90e0570be7cdc541f716540951c3
[I 2024-12-23 00:46:07.879 ServerApp] Skipped non-installed server(s): bash-language-server, dockerfile-language-server-nodejs, javascript-typescript-langserver, jedi-language-server, julia-language-server, pyright, python-language-server, python-lsp-server, r-languageserver, sql-language-server, texlab, typescript-language-server, unified-language-server, vscode-css-languageserver-bin, vscode-html-languageserver-bin, vscode-json-languageserver-bin, yaml-language-server
[W 2024-12-23 00:46:09.455 LabApp] Could not determine jupyterlab build status without nodejs
[W 2024-12-23 00:46:09.624 ServerApp] 404 GET /api/contents/Untitled.ipynb?content=0&hash=0&1734943569620 (d3cd70ea95e74674a730088af5a0fc90@::1) 1.27ms referer=http://localhost:8888/lab
[W 2024-12-23 00:46:09.625 ServerApp] 404 GET /api/contents/Untitled.ipynb?content=0&hash=0&1734943569620 (::1): No such file or directory: Untitled.ipynb
[W 2024-12-23 00:46:09.626 ServerApp] 404 GET /api/contents/Untitled.ipynb?content=0&hash=0&1734943569621 (d3cd70ea95e74674a730088af5a0fc90@::1) 1.13ms referer=http://localhost:8888/lab
[W 2024-12-23 00:46:09.627 ServerApp] 404 GET /api/contents/Untitled.ipynb?content=0&hash=0&1734943569621 (::1): No such file or directory: Untitled.ipynb
[I 2024-12-23 00:46:13.523 ServerApp] Creating new notebook in
pixi-kernel INFO: JupyterLab provided this value for cwd: C:\Users\seembha\Projects\Python\test\src\test
pixi-kernel INFO: The current working directory is C:\Users\seembha\Projects\Python\test\src\test
pixi-kernel INFO: pixi info stderr:
pixi-kernel INFO: pixi info stdout: {
  "platform": "win-64",
  "virtual_packages": [
    "__win=0=0",
    "__archspec=1=x86_64_v4"
  ],
  "version": "0.39.0",
  "cache_dir": "C:\\Users\\seembha\\AppData\\Local\\rattler\\cache",
  "cache_size": null,
  "auth_dir": "C:\\Users\\seembha\\.rattler\\credentials.json",
  "global_info": {
    "bin_dir": "C:\\Users\\seembha\\.pixi\\bin",
    "env_dir": "C:\\Users\\seembha\\.pixi\\envs",
    "manifest": "C:\\Users\\seembha\\.pixi\\manifests\\pixi-global.toml"
  },
  "project_info": {
    "name": "test",
    "manifest_path": "C:\\Users\\seembha\\Projects\\Python\\test\\pyproject.toml",
    "last_updated": "23-12-2024 00:09:30",
    "pixi_folder_size": null,
    "version": "0.1.0"
  },
  "environments_info": [
    {
      "name": "default",
      "features": [
        "default"
      ],
      "solve_group": null,
      "environment_size": null,
      "dependencies": [
        "python",
        "ipykernel"
      ],
      "pypi_dependencies": [
        "test"
      ],
      "platforms": [
        "win-64"
      ],
      "tasks": [],
      "channels": [
        "conda-forge"
      ],
      "prefix": "C:\\Users\\seembha\\Projects\\Python\\test\\.pixi\\envs\\default"
    }
  ],
  "config_locations": []
}

pixi-kernel INFO: Launching Python (Pixi): {'argv': ['pixi', 'run', 'python', '-m', 'ipykernel_launcher', '-f', '{connection_file}'], 'env': {}, 'display_name': 'Python (Pixi)', 'language': 'python', 'interrupt_mode': 'signal', 'metadata': {'debugger': True, 'pixi-kernel': {'required-package': 'ipykernel'}, 'kernel_provisioner': {'provisioner_name': 'pixi-kernel-provisioner', 'config': {}}}}
[I 2024-12-23 00:46:14.597 ServerApp] Kernel started: 3d3442cc-a9cb-4487-9311-2dfaa12105ba
[I 2024-12-23 00:46:18.884 ServerApp] Connecting to kernel 3d3442cc-a9cb-4487-9311-2dfaa12105ba.
[I 2024-12-23 00:46:18.895 ServerApp] Connecting to kernel 3d3442cc-a9cb-4487-9311-2dfaa12105ba.
[I 2024-12-23 00:46:18.913 ServerApp] Connecting to kernel 3d3442cc-a9cb-4487-9311-2dfaa12105ba.
[W 2024-12-23 00:46:19.017 ServerApp] Got events for closed stream <zmq.eventloop.zmqstream.ZMQStream object at 0x00000257EBB242D0>
[I 2024-12-23 00:48:13.688 ServerApp] Saving file at /Untitled.ipynb
```

**Comment by renan-r-santos** [comment]
Thank you for the bug report. I was able to reproduce it and make a fix for it in v0.5.4. Let me know if you still have issues.

**Comment by seembha** [comment]
v0.5.4 fixes the issue. Thank you for the lightning fast bugfix and this awesome utility. Best regards
